### PR TITLE
cifer: fix build for Linux

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

https://github.com/Homebrew/homebrew-core/runs/3153451007?check_suite_focus=true
```
==> make prefix=/home/linuxbrew/.linuxbrew/Cellar/cifer/1.2.0 CC=gcc-5 CFLAGS= LDFLAGS= install
[ -d /home/linuxbrew/.linuxbrew/Cellar/cifer/1.2.0/share/man/man1 ] || install -d /home/linuxbrew/.linuxbrew/Cellar/cifer/1.2.0/share/man/man1
gcc-5 -DVERSION="\"1.2.0\"" -c  src/polybius.c -o src/polybius.o
gcc-5 -DVERSION="\"1.2.0\"" -c  src/actions_io.c -o src/actions_io.o
In file included from src/polybius.c:19:0:
src/stdinc.h:39:31: fatal error: readline/readline.h: No such file or directory
compilation terminated.
Makefile:60: recipe for target 'src/polybius.o' failed
make: *** [src/polybius.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from src/actions_io.c:19:0:
src/stdinc.h:39:31: fatal error: readline/readline.h: No such file or directory
compilation terminated.
Makefile:60: recipe for target 'src/actions_io.o' failed
make: *** [src/actions_io.o] Error 1
```

### CODE REVIEW

File: Formula/cifer.rb [hunk 1]
```diff
@@ -16,6 +16,13 @@ class Cifer < Formula
     sha256 cellar: :any_skip_relocation, yosemite:      "bde7d97d9ef2a07c481ff8c5ec717fb2ec455fdef864db2a1a7b3056aa1934d2"
   end
 
+  on_linux do
+    depends_on "readline"
+
+    # Fix order of linker flags for GCC
+    patch :DATA
```
Comment (iMichka) [inline Formula/cifer.rb:23]: Ideally this should be reported upstream.

Comment (cho-m) [reply Formula/cifer.rb:23]: Not sure if the project is maintained. Last update was in 2009.

Code base is still only on Google Code, which was archived back in 2016.
I don't even know if there is a way to report issues anymore

Comment (iMichka) [reply Formula/cifer.rb:23]: Let's merge this as-is then, as long as it builds. We will probably deprecate this formula one day if it get's too complicated to maintain.

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
https://github.com/Homebrew/homebrew-core/actions/runs/1065301830

**Pull request merged**